### PR TITLE
mta workflow for MTA v7.x, add migration wave Jira related functionality

### PR DESCRIPTION
**Pull request by rhkp**
Hi @masayag @pkliczewski 
Please review and approve or please provide feedback for the changes.
Thanks

### CODE REVIEW

File: mta/README.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+- For MTA v7.0.x, refer to v7.0.x specific version [here](./v7.0.2/openshift-local-setup/readme.md)
+
+# Prerequisites
+* An OpenShift cluster available with MTA Operator 7.x installed
+* The MTA Hub instance with configured [Jira Tracker](https://access.redhat.com/documentation/en-us/migration_toolkit_for_applications/7.0/html/user_interface_guide/creating-configuring-jira-connection#doc-wrapper)
+
+# Inputs
+- `repositoryUrl` [mandatory] - the git repo url to examine
+- `exportToIssueManager` [mandatory] - if true creates migration wave in Jira instance
+- `migrationStartDt` [conditional] - Must be provided, if `exportToIssueManager="true"`. e.g. "2024-07-01T00:00:00Z"
```
[the remark this answers is not in the document]
Comment (rhkp) [reply mta/README.md:21]: Thanks, done.

File: mta/v6.2/openshift-local-setup/envup.sh [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        break
+    else
+        echo "Waiting for: MTA UI Route"
+        sleep 30
+    fi
+done
+
+oc apply -f keycloak-route.yaml
+oc get route
+KEYCLOAK_USER=$(oc get secret credential-mta-rhsso --template={{.data.ADMIN_USERNAME}} | base64 -d)
```
[the remark this answers is not in the document]
Comment (rhkp) [reply mta/v6.2/openshift-local-setup/envup.sh:61]: No, the command works in openshift-mta namespace and addresses the specific `credential-mta-rhsso` OpenShift secret.

File: mta/v7.0.2/src/main/resources/mta.sw.yaml [hunk 3]
```diff
@@ -0,0 +1,374 @@
...
+    transition: report
+  - name: report
+    type: operation
+    stateDataFilter:
+      output: '{result: (.taskgroup.tasks[0]| "report will be ready here -> " + $SECRET.mta.url + "/hub/applications/" + (.application.id|tostring) + "/bucket" + .data.output), repositoryURL: .repositoryURL}'
+    transition: GetWorkflowOptions
+  - name: GetWorkflowOptions
+    type: operation
+    stateDataFilter:
+      output: '{workflowOptions: {migrationOptions: [{id: "m2k", name: "Move2Kube workflow"}]}, repositoryURL: .repositoryURL}'
```
[the remark this answers is not in the document]
Comment (rhkp) [reply mta/v7.0.2/src/main/resources/mta.sw.yaml:359]: The report URL is available in the `report` step, just before the last `m2k` step, as shown in this screenshot.
image

File: mta/v6.2/src/main/resources/mta.sw.yaml [hunk 4]
```diff
@@ -0,0 +1,374 @@
...
+          refName: getTrackers
+        actionDataFilter:
+          toStateData: ".trackers"
+    transition: TrackersExist
+  - name: TrackersExist
+    type: switch
+    dataConditions:
+      - condition: (.trackers != null)
+        transition: GetTracker
+      - condition: (.trackers == null)
```
Comment (dmartinol) [inline mta/v6.2/src/main/resources/mta.sw.yaml:195]: I think all these conditions to check "== null" and transitioninig to `NotifyFailureBackstage` could be removed as this is already the default transition. Just to simplify a bit the flow

File: mta/v7.0.2/cleanup.sh [hunk 5]
```diff
@@ -0,0 +1,20 @@
...zc2Vzc21lbnRzOmdldCBhbmFseXNlczpnZXQgcHJvZmlsZSBhcmNoZXR5cGVzOnBvc3QgYnVzaW5lc3NzZXJ2aWNlczpwdXQgYXBwbGljYXRpb25zLmZhY3RzOmRlbGV0ZSBhcmNoZXR5cGVzOmdldCBzdGFrZWhvbGRlcnM6cG9zdCBqb2JmdW5jdGlvbnM6cHV0IHN0YWtlaG9sZGVyZ3JvdXBzOmRlbGV0ZSBhbmFseXNlczpkZWxldGUgYXBwbGljYXRpb25zOnB1dCBhcHBsaWNhdGlvbnMudGFnczpkZWxldGUgaW1wb3J0czpnZXQgdGFnczpkZWxldGUgcXVlc3Rpb25uYWlyZXM6cG9zdCB0aWNrZXRzOnBvc3QgcmV2aWV3czpwb3N0IGFwcGxpY2F0aW9ucy5hbmFseXNlczpwb3N0IHRyYWNrZXJzOmdldCBqb2JmdW5jdGlvbnM6Z2V0IGFwcGxpY2F0aW9ucy5zdGFrZWhvbGRlcnM6cHV0IHJldmlld3M6Z2V0IHRyYWNrZXJzOmRlbGV0ZSBzdGFrZWhvbGRlcnM6cHV0IHRhc2tzLmJ1Y2tldDpwdXQgdGFza3M6cHV0IHRyYWNrZXJzOnB1dCBzdGFrZWhvbGRlcnM6ZGVsZXRlIGJ1Y2tldHM6Z2V0IGFwcGxpY2F0aW9ucy5idWNrZXQ6ZGVsZXRlIGFkZG9uczpwb3N0Iiwic2lkIjoiZjZhMmM4NDItNWVhNC00NTZlLWJiMjItYjEwMTUwYmU0MDg2IiwiZW1haWxfdmVyaWZpZWQiOmZhbHNlLCJwcmVmZXJyZWRfdXNlcm5hbWUiOiJhZG1pbiJ9.EDmK6-5sdmyhgnorg4MZFRryueG3bDDz4ljm9fPzbuMo-1tccyg0Gyl_YMM43SvWMzF0SsTmW-pjo2dg8IEWSmGYF6j_xXz0KAQE_Rxs6w5NVyZQUGgrUjvxgxUsmkmQdjA4qvneImCuYKttcnbnnskAsiZ3B7qgrGC7XAHknrwpdCSYf-EItQ-bV0-G34rvs1TClt9RnvQa26a2RNE6Rrw5-GYD94VSPL0xKbL0OUSW_lofyQUO-pAtvUCnN7_YdAvWzPzlgCsjmOMFY_R4Bg7-TJyJDefOft9XEpcGXDsdD6rguldxZJpklaQ2WnzkyHMfxjjIyiElcLxapVJPig
```
Comment (dmartinol) [inline mta/v7.0.2/cleanup.sh:1]: shouldn't this AUTH_TOKEN be externalized and documented instead?

Comment (rhkp) [reply mta/v7.0.2/cleanup.sh:1]: This file is not expected to be part of the repo, it is only for local use. This will be deleted.

**Comment by dmartinol** [comment]
> @masayag @dmartinol, Thanks for the valuable comments.
> 
> Please advise of anymore changes or please approve and merge.

LGTM, pls wait for approval from @masayag (but it's w/e today, IIRC)

File: mta/README.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
+# MTA - migration analysis workflow
+
+# Synopsis
+This workflow is an assessment workflow type, that invokes an application analysis workflow using [MTA][1]
+and returns the [move2kube][3] workflow reference, to run next if the analysis is considered to be successful.
+
+Users are encouraged to use this workflow as self-service alternative for interacting with the MTA UI. Instead of running
+a mass-migration of project from a managed place, the project stakeholders can use this (or automation) to regularly check
+the cloud-readiness compatibility of their code.
```
Comment (masayag) [inline mta/README.md:9]: pls include a brief description of the Jira integration and the migration wave feature to identify this workflow from the other MTA we have in https://github.com/parodos-dev/serverless-workflows/tree/main/mta.

Comment (rhkp) [reply mta/README.md:17]: @masayag Charts are in a different `serverless-workflows-config` repo, so is there specific config that you are referring to? Is there a specific example in other workflows that I could follow?

Comment (masayag) [reply mta/README.md:17]: @rhkp yes, eventually, we'd like to transfer this workflow into a workflow that can be consumed by the QE and by users as with the existing MTA workflow.
The process we have taken so far for developing a workflow that can be productive is as follows:

1. Develop the workflow in the current repository as an example. This lets us get the experience and test variations of the solution until we're satisfied with it.
2. Move the workflow to the https://github.com/parodos-dev/serverless-workflows which has a pipeline to productive a workflow. An example of a workflow that pipeline and e2e tests are enabled is the MTA. See 
    1. The workflow definition: https://github.com/parodos-dev/serverless-workflows/tree/main/mta
    2. The workflow pipeline that builds the image https://github.com/parodos-dev/serverless-workflows/blob/main/.github/workflows/mta.yml
    3. e2e tests - responsible for deploying all of the resources required for testing the workflow on kind: https://github.com/parodos-dev/serverless-workflows/blob/main/.github/workflows/mta-e2e.yaml#L135
3. Creating a repository for the workflow image in Quay.io, e.g. https://quay.io/repository/orchestrator/serverless-workflow-mta 
4. The built and generated artifacts are pushed by the pipeline from the previous bullet to https://github.com/parodos-dev/serverless-workflows-config/tree/main/charts/mta (the name may vary based on the workflow name). Next, the templates folder is being extended with the required for the workflow to run (e.g. MTA operator and its configuration). We deliver the workflows as a helm chart. The resources that aren't generated by the pipeline are pushed directly to this repository. 
There is a summary of the things here, but if you find it lacks information, let's improve it: https://github.com/parodos-dev/serverless-workflows/tree/main?tab=readme-ov-file#to-introduce-a-new-workflow
5. Add a page for the new workflow in the gh-pages branch, e.g. https://github.com/parodos-dev/serverless-workflows-config/blob/gh-pages/docs/mta/README.md, and add a reference to it from https://github.com/parodos-dev/serverless-workflows-config/tree/gh-pages/docs#installation

In the end, we'd like to be able to use helm to install the workflow.

Comment (rhkp) [reply mta/README.md:17]: Hi @masayag,
Thanks for a great detailed reply. 

Assuming the current version is good enough, I will start migrating the current version with step 2: moving it workflow to `serverless-workflows` repo. 

Also if the step 1 for this workflow looks okay, would it be a good idea that we merge/close this PR?

Comment (rhkp) [reply mta/README.md:17]: Hi @masayag,
Thanks for a great detailed reply. 

Assuming the current version is good enough, I will start migrating the current version with step 2: moving it workflow to `serverless-workflows` repo. 

Also for the step 1 if there are no further comments from the reviewers, would it be a good idea that we merge/close this PR too?

**Review by masayag — APPROVED** [review]

File: mta/v7.0.2/src/main/resources/mta.sw.yaml [hunk 7]
```diff
@@ -0,0 +1,359 @@
...
+            message: '"MTA analysis is completed and the report for " + .application.repository.url + " is ready."'
+            origin: "MTA Analysis Workflow"
+            topic: "MTA Analysis Workflow"
+            targetUsers: if .backstageUser != null then [.backstageUser] else [] end
+            targetGroups: if .backstageGroup != null then [.backstageGroup] else [] end
+            actions:
+              - title: "View report"
+                url: "$SECRET.mta.url + \"/applications/\" + (.application.id|tostring)"
+    transition: report
+  - name: report
```
Comment (masayag) [inline mta/v7.0.2/src/main/resources/mta.sw.yaml:341]: @rhkp will it be possible to include also the jira issue created by MTA in the result?

Comment (rhkp) [reply mta/v7.0.2/src/main/resources/mta.sw.yaml:341]: Hi @masayag Sure, I will add this into the next commit. Thanks for the feedback.

Comment (rhkp) [reply mta/v7.0.2/src/main/resources/mta.sw.yaml:341]: Hi @masayag,
I have added the Jira Issue URL in the output as shown in the screenshot. This is in v7.x and I will add the similar change in v6.2 as well.

image

Comment (masayag) [reply mta/v7.0.2/src/main/resources/mta.sw.yaml:341]: Thank you @rhkp

**Pull request merged**